src/apps/deskflow-gui/MainWindow.cpp

```diff
@@ -663,44 +701,47 @@ void MainWindow::checkConnected(const QString &line)
 
 void MainWindow::checkFingerprint(const QString &line)
 {
-  static const QRegularExpression re(".*server fingerprint: ([A-F0-9:]+)");
+  static const QRegularExpression re(R"(.*server fingerprint: \(SHA1\) ([A-F0-9:]+) \(SHA256\) ([A-F0-9:]+))");
   auto match = re.match(line);
   if (!match.hasMatch()) {
     return;
   }
 
-  auto fingerprint = match.captured(1);
-  if (TlsFingerprint::trustedServers().isTrusted(fingerprint)) {
-    return;
-  }
+  const deskflow::FingerprintData sha1 = {
+      deskflow::fingerprintTypeToString(deskflow::FingerprintType::SHA1),
+      deskflow::string::fromHex(match.captured(1).toStdString())
+  };
 
-  static bool messageBoxAlreadyShown = false;
-
-  if (!messageBoxAlreadyShown) {
-    m_coreProcess.stop();
-
-    messageBoxAlreadyShown = true;
-    QMessageBox::StandardButton fingerprintReply = QMessageBox::information(
-        this, tr("Security question"),
-        tr("<p>You are connecting to a server.</p>"
-           "<p>Here is it's TLS fingerprint:</p>"
-           "<p>%1</p>"
-           "<p>Compare this fingerprint to the one on your server's screen. "
-           "If the two don't match exactly, then it's probably not the server "
-           "you're expecting (it could be a malicious user).</p>"
-           "<p>Do you want to trust this fingerprint for future "
-           "connections? If you don't, a connection cannot be made.</p>")
-            .arg(fingerprint),
-        QMessageBox::Yes | QMessageBox::No
-    );
+  const deskflow::FingerprintData sha256 = {
+      deskflow::fingerprintTypeToString(deskflow::FingerprintType::SHA256),
+      deskflow::string::fromHex(match.captured(2).toStdString())
+  };
 
-    if (fingerprintReply == QMessageBox::Yes) {
-      // start core process again after trusting fingerprint.
-      TlsFingerprint::trustedServers().trust(fingerprint);
-      m_coreProcess.start();
+  const bool hasSHA256 = !match.captured(2).isEmpty();
+
+  auto localPath = QStringLiteral("%1/%2").arg(getTlsPath(), kFingerprintTrustedServersFilename).toStdString();
+  deskflow::FingerprintDatabase db;
+  db.read(localPath);
+  if (hasSHA256) {
+    if (db.isTrusted(sha256)) {
+      return;
+    }
+  } else {
+    if (db.isTrusted(sha1)) {
+      return;
     }
+  }
```

Comment:
Pretty sure the logic is the same in this suggestion. Normally, I'd want to see a unit test here to prove it.

```suggestion
  if (hasSHA256 && db.isTrusted(sha256)) {
    return;
  } else if (db.isTrusted(sha1)) {
    return;
  }
```